htdocs/accountancy/bookkeeping/export.php

```diff
@@ -1190,7 +1190,7 @@
 			require_once DOL_DOCUMENT_ROOT.'/compta/facture/class/facture.class.php';
 			$objectstatic = new Facture($db);
 			$objectstatic->fetch($line->fk_doc);
-			//$modulepart = 'facture';
+			$modulepart = 'facture'; // or use $objectstatic->element
```

Comment:
The variable $modulepart is defined into a if but not used inside this if. This make me think it is defined at wrong place.
Where do you need to use it ?

Reply:
@eldy maybe because of a `global $modulepart;` into completeFileArrayWithDatabaseInfo function used to build the zip file i think but it's too hard to follow all stack call to be sure

Reply:
so @josett225 could you please help us to find why that variable set will change the result ?

maybe a dolibarr in debug mode, collect the log with $modulepart in comments as official dolibarr sources, then apply your patch and redo the export, compare the log and send us more informations

thanks

Reply:
I will do.

Reply:
Hi @eldy and @rycks 
After diffing more in several versions, I found out we could remove $modulepart and replace it $objectstatic->element everywhere.
I changed the code accordingly and remove any mention of $modulepart

Regards
Jose